table/tables/cache.go

```diff
@@ -273,3 +289,60 @@ func (c *cachedTable) renewLease(ts uint64, data *cacheData, leaseDuration time.
 		TestMockRenewLeaseABA2 <- struct{}{}
 	})
 }
+
+const cacheTableWriteLease = 5 * time.Second
+
+func (c *cachedTable) WriteLockAndKeepAlive(ctx context.Context, ts uint64, exit chan struct{}, leasePtr *uint64, wg chan error) {
+	writeLockLease, err := c.lockForWrite(ctx, ts)
+	atomic.StoreUint64(leasePtr, writeLockLease)
+	wg <- err
+	if err != nil {
+		logutil.Logger(ctx).Warn("[cached table] lock for write lock fail", zap.Error(err))
+		return
+	}
+
+	t := time.NewTicker(cacheTableWriteLease)
+	defer t.Stop()
+	for {
+		select {
+		case <-t.C:
+			if err := c.renew(ctx, leasePtr); err != nil {
+				logutil.Logger(ctx).Warn("[cached table] renew write lock lease fail", zap.Error(err))
+				return
+			}
+		case <-exit:
+			return
+		}
+	}
+}
+
+func (c *cachedTable) renew(ctx context.Context, leasePtr *uint64) error {
+	oldLease := atomic.LoadUint64(leasePtr)
+	physicalTime := oracle.GetTimeFromTS(oldLease)
+	newLease := oracle.GoTimeToTS(physicalTime.Add(cacheTableWriteLease))
+
+	c.tokenLimit <- struct{}{}
+	defer func() { <-c.tokenLimit }()
+
+	succ, err := c.handle.RenewWriteLease(ctx, c.Meta().ID, newLease)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if succ {
+		atomic.StoreUint64(leasePtr, newLease)
+	}
+	return nil
+}
+
+func (c *cachedTable) lockForWrite(ctx context.Context, ts uint64) (uint64, error) {
```

Comment:
seems `ts` is not used. It can be removed for this method and `WriteLockAndKeepAlive` can also remove `ts` too